code/obj/item/swingsign.dm

```diff
@@ -0,0 +1,197 @@
+
+	New()
+		..()
+		if(message)
+			setmessage(message)
+
+	proc/setmessage(var/newmessage)
+		message = newmessage
+		if(message == "")
+			desc = defaultdesc
+		else
+			desc = descpreamble + message + "</div>"
+		UpdateIcon()
+
+	update_icon(...)
+		if(message == "")
+			icon_state = "blank"
+		else
+			icon_state = "written"
+
+	proc/fold()
+		var/obj/item/swingsignfolded/newSwingsign = new/obj/item/swingsignfolded(src.loc)
+
+		if (src.material)
+			newSwingsign.setMaterial(src.material)
+		//if (src.icon_state)
+		//	newSwingsign.icon_state = "folded"
+		if (src.message)
+			newSwingsign.message = src.message
+
+		qdel(src)
+
+	attack_hand(mob/user)
+		if (!isliving(user)) return
+		if(!secured)
+			user.visible_message("<b>[user.name] folds [src].</b>")
+			playsound(src, 'sound/impact_sounds/Clock_slap.ogg',50,1)
+			fold()
+		else
+			boutput(user, "<span alert='notice'>Sign is too tightly secured to fold!</span>")
+		return
+
+	attackby(obj/item/W, mob/user)
+		if(istype(W, /obj/item/pen))
+			ui_interact(user)
+			return
+		else if (isscrewingtool(W))
+			if(secured)
+				boutput(user, "<span class='notice'>You unsecure the swing sign.</span>")
+				anchored = UNANCHORED
+			else
+				boutput(user, "<span class='notice'>You secure the swing sign.</span>")
+				anchored = ANCHORED
+			playsound(src, 'sound/items/Screwdriver.ogg', 50, 1)
+			secured = !secured
+			return
+		..()
+
+	hitby(atom/movable/M, datum/thrown_thing/thr)
+		if (isliving(M) && M.throwing)
+			var/mob/living/L = M
+			var/area/T = get_area(src)
+			if(T?.sanctuary)
+				return
+			random_brute_damage(M, rand(1,maxcrashdamage),1)
+			L.do_disorient(10, 0, 1 SECOND, 0, disorient = 0, stack_stuns = FALSE)
+			L.force_laydown_standup()
+			//L.changeStatus("stunned", 1 SECONDS)
+			src.visible_message("<b><font color=red>[M] is people-stopped by [src]!</font></b>")
+			playsound(src, 'sound/impact_sounds/Wood_Hit_1.ogg',50,1)
+			fold() //Change to item
+			return
+		..()
+
+
+	suicide(var/mob/user as mob)
+		if (!src.user_can_suicide(user))
+			return 0
+		user.visible_message("<span class='alert'><b>[user] puts [his_or_her(user)] head between [src]'s legs and clamps them shut!</b></span>")
+		user.TakeDamage("head", 250, 0)
+		playsound(src, 'sound/items/woodbat.ogg',50,1)
+		fold()
+		return 1
+
+
+/obj/swingsign/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SwingSign")
+		ui.open()
+
+/obj/swingsign/ui_data(mob/user)
+	. = list(
+		"message" = newline_html_decode(src.message),//Decodes the message for the UI component and replaces all <br> with \n (text.dm)
+		"maxRows" = src.maxmessagerows,
+		"maxCols" = src.maxmessagecols
+	)
+
+/obj/swingsign/ui_act(action, params)
+	. = ..()
+	if (.)
+		return
+	if(action == "save_message")
+		var/new_message = newline_html_encode(params["message"])//Encodes the message for safety and replaces all \n with <br> (text.dm)
```

Comment:
you still want a copytext here to enforce character limit. The client side TGUI JS is not trustworthy enough to enforce.